pkg/discovery/util/node_util.go

```diff
@@ -29,6 +29,7 @@ const (
 	// NodeLabelArch and NodeLabelArchBeta specify the arch of a node
 	NodeLabelArch     = "kubernetes.io/arch"
 	NodeLabelArchBeta = "beta.kubernetes.io/arch"
+	NodePoolAKS       = "agentpool"
```

Comment:
I am assuming there is a reason we only have AKS `agentpool` in the labels list as of now?

Reply:
added node pool labels for GKE and EKS

Reply:
@pallavidn `kubernetes.io/arch` and `beta.kubernetes.io/arch` are generic labels meant for architecture of nodes. 

There would be different label keys used for EKS and GKE to indicate their corresponding node pools. For example in EKS the labels should start with `eks.amazonaws.com` as listed [here](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html). GKE uses `cloud.google.com/gke-nodepool` as listed [here](https://cloud.google.com/kubernetes-engine/docs/concepts/node-pools).

However, I think we can add the EKS labels probably after we have clusters which enable the pools, which I don't know if we have any (as the documentation does not list the exact key used for the same).
cc @evat-pm

Reply:
@irfanurrehman, the arch related labels are part of existing code.
The labels I added are below - see the updated file.
@evat-pm and I created new node pool groups in GKE and EKS and these are the labels based from those node pools.
GKE uses label - "cloud.google.com/gke-nodepool" and EKS has eks.amazonaws.com but older versions of  EKS clusters also have node pools with a different key.
     ```
       // AKS Node pool label
	NodePoolAKS = "agentpool"
	// GKE Node pool label
	NodePoolGKE = "cloud.google.com/gke-nodepool"
	// EKS Node pool label
	NodePoolEKSIdentifier = "/nodegroup" //alpha.eksctl.io/nodegroup-name or eks.amazonaws.com/nodegroup
```